<a href="https://colab.research.google.com/github/DanielCadavid/InteligenciaArtificial/blob/main/aprendizajesupervisado.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Aprendizaje Supervisado


> A continuacion se presenta una implementacion de Algoritmos de clasificacion

##Regresion Lineal
Instalacion de Libreria



In [1]:
!pip3 install mglearn

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 581.4/581.4 kB 7.0 MB/s eta 0:00:00


Para demostrar el funcionamiento de la regresion lineal, se trae un set de datos "wave"

In [11]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import mglearn
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
X, y = mglearn.datasets.make_wave(n_samples=60)
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)
lr = LinearRegression().fit(X_train, y_train)
print("lr.coef_: {}".format(lr.coef_))
print("lr.intercept_: {}".format(lr.intercept_))

lr.coef_: [0.39390555]
lr.intercept_: -0.031804343026759746


La variable "lr.coef" posee el valor de los pesos en un arreglo de tipo NumPy, mientras "lr.interceot" almacena el valor del intercepto con el eje y



>Ahora se mostraran los resultados de entrenamiento y prueba

In [12]:
print("Training set score: {:.2f}".format(lr.score(X_train, y_train)))
print("Test set score: {:.2f}".format(lr.score(X_test, y_test)))

Training set score: 0.67
Test set score: 0.66


La regresion lineal tiene como objetivo encontrar los parametros de las variables que minimicen el error cuadratico entre las predicciones y el objetivo de la regresion.
El 66% no es un buen porcentaje en los entrenamientos, ademas se puede observar una minima diferencia frente a las pruebas. Lo que indica un desajuste o falta de datos, Por lo que vamos a ver un segundo ejemplo con un set de datos mas grande y con mas caracteristicas